addons/binding/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/handler/SatelThingHandler.java

```diff
@@ -0,0 +1,179 @@
+        if (event instanceof ConnectionStatusEvent) {
+            ConnectionStatusEvent statusEvent = (ConnectionStatusEvent) event;
+            // we have just connected, force refresh state
+            if (statusEvent.isConnected()) {
+                getThing().setProperty(PROPERTY_REQUIRES_REFRESH, Boolean.toString(true));
+            }
+        } else if (event instanceof NewStatesEvent) {
+            // refresh all states that have changed
+            for (SatelCommand command : getRefreshCommands((NewStatesEvent) event)) {
+                bridgeHandler.sendCommand(command, true);
+            }
+            if (getThing().getStatus() != ThingStatus.ONLINE) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+        } else if (event instanceof IntegraStateEvent) {
+            // update thing's state unless it should accept commands only
+            IntegraStateEvent stateEvent = (IntegraStateEvent) event;
+            if (thingConfig.isCommandOnly()) {
+                return;
+            }
+            for (Channel channel : getThing().getChannels()) {
+                ChannelUID channelUID = channel.getUID();
+                StateType stateType = getStateType(channelUID.getId());
+                if (stateType != null && stateEvent.hasDataForState(stateType)) {
+                    int bitNbr = thingConfig.getId() - 1;
+                    boolean invertState = thingConfig.isStateInverted();
+                    updateSwitch(channelUID, stateEvent.isSet(stateType, bitNbr) ^ invertState);
+                }
+            }
+        }
+    }
+
+    protected abstract SatelCommand convertCommand(ChannelUID channel, Command command);
+
+    protected abstract StateType getStateType(String channelId);
+
+    protected Channel getChannel(StateType stateType) {
+        String channelId = stateType.toString().toLowerCase();
+        Channel channel = getThing().getChannel(channelId);
+        if (channel == null) {
+            logger.warn("Missing channel for {}", stateType);
+        }
+        return channel;
+    }
+
+    protected Collection<SatelCommand> getRefreshCommands(NewStatesEvent event) {
+        Collection<SatelCommand> result = new LinkedList<>();
+        boolean requiresRefresh = requiresRefresh();
+        for (Channel channel : getThing().getChannels()) {
+            boolean isChannelLinked = !linkRegistry.getLinks(channel.getUID()).isEmpty();
+            if (isChannelLinked) {
+                StateType stateType = getStateType(channel.getUID().getId());
+                if (requiresRefresh || event.isNew(stateType.getRefreshCommand())) {
+                    result.add(new IntegraStateCommand(stateType, bridgeHandler.getIntegraType().hasExtPayload()));
+                }
+            }
+        }
+        return result;
+    }
+
+    protected void updateSwitch(String channelID, boolean switchOn) {
+        ChannelUID channelUID = new ChannelUID(this.getThing().getUID(), channelID);
```

Comment:
Note that you can call `updateState()` with the channelID instead of the channelUID as well, so no need to construct the UID here.

Reply:
This helps optimizing things in case channel UID is present. The method you mention does the same - creates UID. In case it is available it does not make sense to create it again.